dev/src/index.ts

```diff
@@ -502,30 +481,55 @@ export class Firestore implements firestore.Firestore {
       this._projectId = settings.projectId;
     }
 
-    if (settings.host !== undefined) {
+    let url: URL | null = null;
+
+    // If the environment variable is set, it should always take precedence
+    // over any user passed in settings.
```

Comment:
If we feel strongly that this is the wrong thing to do, should add a TODO to fix this?

Reply:
I don't see how we would fix it without creating an inconvenient breaking change for developers. Since this is only used for emulators and 1p testing, I think the current behavior isn't "wrong". I mainly wanted to document the priority order to make the code easier to understand.